[Xcode] adds script for updating Settings bundle

KevinVitale opened the pull request:
![duckduckgo-settings](https://user-images.githubusercontent.com/276137/69600785-eb6af380-0fdf-11ea-975b-593734261745.png)
<hr/>
Task/Issue URL: #497 

**Description**:
This adds:
 1. a config file that defines `MARKETING_VERSION`–`Version.xcconfig`;
 2. a settings bundle–`Settings.bundle`;
 3. a script file–`set_version.sh`.

The script file pulls `MARKETING_VERSION` from the config file, and writes it to the settings bundle.

**Steps to test this PR**:
1. Run `chmod +x set_version.sh` to ensure it is executable;
2. Run `set_version.sh 7.33.1`

**Known Issues**:
This script should be run prior to build the app for a release, and the resulting changes committed.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

bwaresiak commented on set_version.sh:
```diff
@@ -0,0 +1,102 @@
+catch {
+  print(error)
+}
+
```
This is pretty nice implementation that uses swift, but seems a bit comlplex. Have you considered using Mac OS tools instead?

Example:
```#!/usr/bin/env bash

if [ -z "$1" ]; then
   echo Usage: ./set_version NUMBER
   echo Example: ./set_version 7.100.1
   exit 1
fi

echo "MARKETING_VERSION = $1" > Configuration/Version.xcconfig
/usr/libexec/PlistBuddy -c "Set :PreferenceSpecifiers:0:DefaultValue $1" DuckDuckGo/Settings.bundle/Root.plist
```

KevinVitale replied:
Ah! _PlistBuddy_ is **so** much simpler—the only caveat being the hard-coded index (which could be easily forgotten about if more things get added in the future).

I ran your implementation and got the same diff output! This is awesome feedback; thanks so much. I'll squash the commits and update the PR.

KevinVitale committed:
[Xcode] adds script for updating Settings bundle

KevinVitale committed:
[Xcode] adds script preventing version changes

Each build target adds a "Run Script Phase", which runs `check_version.sh`. The purpose of this script is detect any overrides of `MARKETING_VERSION` that has been set manually within Xcode.

Version changes should be performed by running the `set_version.sh` script.

bwaresiak reviewed (approved):
Great! This looks good to me, thank you for the contribution. 😄
